oauthdoctor/diag/config.go

```diff
@@ -187,7 +188,9 @@ func (c *ConfigFile) ReplaceConfigFromReader(key, value string, r io.Reader) str
 		langKey := c.GetConfigKeysInLang(key)
 
 		// Found the line with old config key and comment it out
-		if !strings.HasPrefix(trimmedLine, commentChar) && strings.Contains(trimmedLine, langKey) {
+		if c.Lang == "dotnet" && strings.Contains(trimmedLine, langKey) {
+			buf.WriteString("<!-- " + trimmedLine + " -->\n")
```

Comment:
Since you have defined CommentChar in languages why are you using a comment literal here?

Reply:
XML comment has prefix and suffix. I can do  `commentChar + trimmedLine + " -->\n"`

but I think writing it out explicitly is clearer.

Reply:
Not a big deal, but I don't think that it makes it clearer and introduces a dependency if it ever changes.

commentChar + trimmedLine + commentChar + "\n" is clearer to me.